### PR TITLE
plan:  converting plan to pb should not rely on `Column`'s `Position`

### PR DESCRIPTION
**Pull request by winoros**
## What have you changed? (mandatory)

To remove the `FromID` in `expression.Column`. We need to remove the implicative meaning in `expressoin.Column.Position`.

## What is the type of the changes? (mandatory)

- refactor

## How has this PR been tested? (mandatory)

existing tests.

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -1589,7 +1591,7 @@ func buildNoRangeIndexLookUpReader(b *executorBuilder, v *plan.PhysicalIndexLook
 		keepOrder:         is.KeepOrder,
 		desc:              is.Desc,
 		tableRequest:      tableReq,
-		columns:           is.Columns,
+		columns:           ts.Columns,
```
Comment (zz-jason) [inline executor/builder.go]: why change this?

Comment (winoros) [reply executor/builder.go]: Though `IndexScan`' `Columns` is directly copied from `DataSource`, but its actual meaning is not this.
So change to `TableScan`'s `Columns' to make this more clear.

File: plan/plan_to_pb.go [hunk 2]
```diff
@@ -120,12 +120,9 @@ func (p *PhysicalIndexScan) ToPB(ctx sessionctx.Context) (*tipb.Executor, error)
...
 	for _, col := range p.schema.Columns {
 		if col.ID == model.ExtraHandleID {
-			columns = append(columns, &model.ColumnInfo{
-				ID:   model.ExtraHandleID,
-				Name: model.ExtraHandleName,
-			})
+			columns = append(columns, model.NewExtraHandleColInfo())
 		} else {
-			columns = append(columns, tableColumns[col.Position])
+			columns = append(columns, model.FindColumnInfo(tableColumns, col.ColName.L))
```
Comment (coocood) [inline plan/plan_to_pb.go]: How about find column by ID?

Comment (winoros) [reply plan/plan_to_pb.go]: @coocood `IndexScan`'s `Column` in `Schema` is generated from `IndexInfo.IndexColumns` which don't have `ID`. So if we find column by ID here. We need to find its ID by name when create schema.

**Pull request merged**